tests/UnitTests/GT.Core.Tests/Services/GTCompanyServiceTests.cs

```diff
@@ -0,0 +1,561 @@
+		}
+
+		[Fact]
+		public async Task DeleteAsync_ValidInputId_Succeeds()
+		{
+			//	Arrange
+			string inputCompanyId = "Valid companyId";
+
+			var mockLogger = new Mock<ILogger<GTCompanyService>>();
+			var mockRepository = new Mock<IGTGenericRepository<Company>>();
+			var callBackResult = new Company();
+
+			mockRepository
+				.Setup(m => m.Get())
+				.Returns(new List<Company>()
+				{
+					new Company()
+					{
+						Id = inputCompanyId
+					}
+				}.AsQueryable().BuildMock());
+
+			var sut = new GTCompanyService(mockLogger.Object, mockRepository.Object);
+
+			//	Act
+			await sut.DeleteAsync(inputCompanyId);
+
+			//	Assert
+			mockRepository.Verify(m => m.DeleteAsync(It.IsAny<Company>()), Times.Once);
+		}
+
+		[Fact]
+		public async Task ExistsByNameAsync_ValidInputName_Succeeds()
+		{
+			//	Arrange
+			string inputCompanyName = "Valid companyName";
+
+			var mockLogger = new Mock<ILogger<GTCompanyService>>();
+			var mockRepository = new Mock<IGTGenericRepository<Company>>();
+			var callBackResult = new Company();
+
+			mockRepository
+				.Setup(m => m.Get())
+				.Returns(new List<Company>()
+				{
+					new Company()
+					{
+						Name = inputCompanyName
+					}
+				}.AsQueryable().BuildMock());
+
+			var sut = new GTCompanyService(mockLogger.Object, mockRepository.Object);
+
+			//	Act
+			var result = await sut.ExistsByNameAsync(inputCompanyName);
+
+			//	Assert
+			mockRepository.Verify(m => m.Get(), Times.Once);
+
+			result.Should()
+				.BeTrue();
+		}
+
+		[Theory]
+		[InlineData("")]
+		[InlineData(null)]
+		public async Task ExistsByNameAsync_InvalidInputName_Fails(string inputCompanyName)
+		{
+			//	Arrange
+			var mockLogger = new Mock<ILogger<GTCompanyService>>();
+			var mockRepository = new Mock<IGTGenericRepository<Company>>();
+			var callBackResult = new Company();
+
+			var sut = new GTCompanyService(mockLogger.Object, mockRepository.Object);
+
+			//	Act
+			await sut.ExistsByNameAsync(inputCompanyName);
+
+			//	Assert
+			mockRepository.Verify(m => m.Get(), Times.Never);
+		}
+
+		[Fact]
+		public async Task ExistsByNameAsync_ValidNameDoesNotExistInDB_SucceedsButReturnsFalse()
+		{
+			//	Arrange
+			string inputCompanyName = "Valid companyName";
+
+			var mockLogger = new Mock<ILogger<GTCompanyService>>();
+			var mockRepository = new Mock<IGTGenericRepository<Company>>();
+			var callBackResult = new Company();
+
+			mockRepository
+				.Setup(m => m.Get())
+				.Returns(new List<Company>()
+				{
+					new Company()
+					{
+						Name = "Google"
+					}
+				}.AsQueryable().BuildMock());
+
+			var sut = new GTCompanyService(mockLogger.Object, mockRepository.Object);
+
+			//	Act
+			var result = await sut.ExistsByNameAsync(inputCompanyName);
+
+			//	Assert
+			mockRepository.Verify(m => m.Get(), Times.Once);
+
+			result.Should()
+				.BeFalse();
+		}
+
+		[Fact]
+		public async Task GetAllAsync_ThreeEntitiesExistInDB_SucceedsAndReturnsThreeEntities()
```

Comment:
We should add a test for GetAllAsync() which verifies that the mapping of DTOs are what we expect them to be.

Reply:
This needs to be implemented in previous tests as well as I do not believe that we have tested this in other services.